From 765f8344cd33c29f808a32c4b994c9ca562fe745 Mon Sep 17 00:00:00 2001
From: Tor Lillqvist <tml@iki.fi>
Date: Tue, 22 Jun 2010 18:30:13 +0000
Subject: Generate correct grab broken event for WM_KILLFOCUS

WM_KILLFOCUS means that a keyboard grab (not a pointer grab), if any,
has been broken. I don't think this bug has matterd much as gtk
generates a grab-broken-event signal for both keybord and pointer
grabs being broken anyway.
---
diff --git a/gdk/win32/gdkevents-win32.c b/gdk/win32/gdkevents-win32.c
index 3773e4d..23dcd24 100644
--- a/gdk/win32/gdkevents-win32.c
+++ b/gdk/win32/gdkevents-win32.c
@@ -2539,7 +2539,7 @@ gdk_event_translate (MSG  *msg,
       if (_gdk_display->keyboard_grab.window != NULL &&
 	  !GDK_WINDOW_DESTROYED (_gdk_display->keyboard_grab.window))
 	{
-	  generate_grab_broken_event (_gdk_display->keyboard_grab.window, FALSE, NULL);
+	  generate_grab_broken_event (_gdk_display->keyboard_grab.window, TRUE, NULL);
 	}
 
       /* fallthrough */
--
cgit v0.8.3.1
